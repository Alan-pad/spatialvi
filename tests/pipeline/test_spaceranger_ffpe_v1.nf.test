nextflow_pipeline {
    name "Test full workflow including spaceranger"
    script "main.nf"
    tag "pipeline"

    test("spaceranger ffpe v1") {
        when {
            params {
                input  = 'test-datasets/testdata/human-ovarian-cancer-1-standard_v1_ffpe/samplesheet_spaceranger.csv'
                spaceranger_probeset = 'test-datasets/testdata/human-ovarian-cancer-1-standard_v1_ffpe/Visium_Human_Transcriptome_Probe_Set_v1.0_GRCh38-2020-A.csv'
                st_preprocess_min_counts = 5
                st_preprocess_min_genes = 3
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir")).match("software_versions") },
                // data
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/data/st_adata_processed.h5ad").exists() },
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/data/st_adata_norm.h5ad").exists() },
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/data/st_adata_plain.h5ad").exists() },
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/data/st_adata_raw.h5ad").exists() },
                // reports
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/reports/st_clustering.html").text.contains("Saving anndata file for future use") },
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/reports/st_qc_and_normalisation.html").text.contains("Distribution after filtering") },
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/reports/st_spatial_de.html").text.contains("Spatially variable genes") },
                // degs
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/degs/st_spatial_de.csv").exists() },
                // spaceranger
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/outs/web_summary.html").exists() },
                { assert snapshot(
                    path("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/outs/raw_feature_bc_matrix.h5"),
                    path("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/outs/filtered_feature_bc_matrix.h5"),
                )},
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/outs/raw_feature_bc_matrix.h5").exists() },
                { assert file("$outputDir/Visium_FFPE_Human_Ovarian_Cancer/outs/spatial/tissue_positions.csv").exists() }
            )
        }
    }



}
