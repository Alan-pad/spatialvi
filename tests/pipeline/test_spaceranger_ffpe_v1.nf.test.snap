{
    "reports": {
        "content": [
            "st_clustering.html:md5,905395042fa8b8a012718c1dbff39db4",
            "st_qc_and_normalisation.html:md5,f89922b04edd7299e34ba1e43c257000"
        ],
        "timestamp": "2023-06-22T09:30:11+0000"
    },
    "software_versions": {
        "content": [
            "{CUSTOM_DUMPSOFTWAREVERSIONS={python=3.11.0, yaml=6.0}, SPACERANGER_COUNT={spaceranger=2.1.0}, SPACERANGER_UNTAR_REFERENCE={untar=1.30}, ST_CLUSTERING={quarto=1.3.302, scanpy=1.9.3}, ST_QC_AND_NORMALISATION={quarto=1.3.302, scanpy=1.9.3}, ST_READ_DATA={scanpy=1.7.2}, ST_SPATIAL_DE={SpatialDE=, leidenalg=, quarto=1.3.302, scanpy=1.9.3}, Workflow={nf-core/spatialtranscriptomics=1.0dev}}"
        ],
        "timestamp": "2023-06-22T09:30:11+0000"
    }
}
